.github/workflows/check-release.yml

```diff
@@ -17,13 +16,18 @@ jobs:
       image: python:latest
     env:
       GITHUB_PAT: ${{ secrets.INSTALL_GITHUB }}
-    # If the PR is coming from a fork (pull_request_target), ensure it is opened
-    # by "dependabot[bot]". Otherwise, clone it normally.
       CC_API_ORGANISATION: ${{ secrets.CC_API_ORGANISATION }}
       CC_API_KEY: ${{ secrets.CC_API_KEY }}
-    if: |
-        (github.event_name == 'pull_request_target' && github.actor == 'dependabot[bot]') ||
-        (github.event_name != 'pull_request_target' && github.actor != 'dependabot[bot]')
+    if: >
+      (github.event_name != 'pull_request' && !github.event.pull_request.head.repo.fork) ||
+      (github.event_name == 'pull_request' &&
+      (github.event.pull_request.head.repo.fork || github.actor == 'dependabot[bot]'))
```

Comment:
These lines are cryptic. Best to leave a comment telling us what theses lines good for / why we need this snippet.

Note: This file fiddles with configurations and is rather hard to understand for an unexperienced user (like me). Contrary to our usual Python code where we target code readability, we need comments here for clarity.

Reply:
Added